### PR TITLE
Omit Custom Request Guards

### LINKED ISSUE #3: Omit Custom Request Guards
When generating our TypeScript interface we should omit [Custom Request Guards](https://rocket.rs/guide/v0.5/requests/#custom-guards) as the client doesn't need to think about these.

For example, `example-handlers/thread.rs` includes an `AgentService` that is provided to us automatically by Rocket. Since `AgentService` implements `FromRequest`, Rocket has all it needs to construct this from the request itself (usually an user auth header or similar) and so the client doesn't need to include it explicitly in their request. 

~Ideally we'd detect when types implement `FromRequest` and ignore them.~ If that turns out to be overly difficult I suggest we add a CLI parameter for specifying a file containing type exclusions. We could default to skipping types like `Db` and `State`.

### PR DESCRIPTION
**Pull request by AvaterClasher**
Nothing fancy just storing the req guards in a vector and comparing the types against it. If we don't get a exclude type we return a empty vec so that the compiler doesnt panic.

In the base folder -> create `exclusion.txt || exclusion.toml` anything really.

`cargo run generate -i example-handlers -o k7.ts -e exclusion.toml `

closes #3

### CODE REVIEW

**Comment by dteare** [comment]
I don't see how the exclusion list could be a `toml` file. Afaict from the parsing code you're expecting a file of new-line separated strings.

File: src/main.rs [hunk 1]
```diff
@@ -55,11 +55,19 @@ enum Commands {
...
             help = "Output file",
-            default_value = "k7.ts",
             short = 'o',
             long = "output"
         )]
         output_file: String,
+
+        #[clap(
+            required = false,
+            help = "Exclusion file",
```
Comment (dteare) [inline src/main.rs:65]: This help text could be so much more helpful to the user. As far as I know this is _only_ used for excluding parameters from the generated interface, and it's (likely?) only useful for Request Guards. It'd be helpful to touch on these concepts here so the reader can quickly piece together why they would want this.

File: src/main.rs [hunk 2]
```diff
@@ -101,6 +110,14 @@ fn main() -> std::io::Result<()> {
                 }
             }
 
+            let exclusion_list = if let Some(exclude_file) = exclude_file {
+                println!("Loading the Exclusion File : {}", &exclude_file);
+                parse_exclusion_file(&exclude_file)
+            } else {
+                // If no exclusion file is provided, initializing an empty exclusion list so the compiler doesnt panic
```
Comment (dteare) [inline src/main.rs:117]: This comment isn't needed. It just seems to get in the way.

File: src/parser/params_as_comma_seperated.rs [hunk 3]
```diff
@@ -3,22 +3,25 @@ use quote::quote;
...
+/// This function takes a vector of `FnArg` items, typically representing the arguments of a function,
 /// and generates a comma-separated string of parameter names and their types. It handles various
-/// types of function arguments including references, path types, and more.
+/// types of function arguments including references, path types, and more. Additionally, it excludes
+/// types specified in the provided exclusion list.
 ///
 /// # Arguments
 ///
-/// * `args` - A vector of FnArg items representing the arguments of a function.
+/// * `args` - A vector of `FnArg` items representing the arguments of a function.
```
Comment (dteare) [inline src/parser/params_as_comma_seperated.rs:17]: I struggle with `Arguments` comments as they need to be maintained and personally I find just reading the code `args: Vec<FnArg>` directly is faster and easier to understand. To me this practice is really just an artifact from earlier languages that lacked strict typing and lack value in Rust. Especially for internal modules.

File: src/parser/exclusion_parser.rs [hunk 4]
```diff
@@ -0,0 +1,15 @@
+use std::fs;
+
+/// Parses the exclusion file and returns a vector of types to exclude.
+pub fn parse_exclusion_file(exclusion_file_path: &String) -> Vec<String> {
```
Comment (dteare) [inline src/parser/exclusion_parser.rs:4]: As a user of `rocket-ts` I would expect an error if I provided a file that didn't exist and the program should halt. The parser can't be responsible for that so a `Result` type is really needed here.

File: src/parser/exclusion_parser.rs [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    if let Ok(contents) = fs::read_to_string(exclusion_file_path) {
+        contents.lines().map(|line| line.trim().to_string()).collect()
```
Comment (dteare) [inline src/parser/exclusion_parser.rs:6]: It really feels like there should be some sort of validation of the contents of the file. Maybe I'm over thinking it though.

File: src/parser/exclusion_parser.rs [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+    } else {
+        Vec::new()
```
Comment (dteare) [inline src/parser/exclusion_parser.rs:8]: With a `Result` type this wouldn't be needed.

**Review by dteare — CHANGES_REQUESTED** [review]
I left some comments.

File: src/main.rs (hunk 1 above)
Comment (AvaterClasher) [reply src/main.rs]: Left helpful message for the user.

File: src/main.rs (hunk 2 above)
Comment (AvaterClasher) [reply src/main.rs:116]: Removed

File: src/parser/exclusion_parser.rs (hunk 4 above)
Comment (AvaterClasher) [reply src/parser/exclusion_parser.rs:4]: Added a result type to handle if the file didnt exist. the programs stop execution if that happens.

File: src/parser/exclusion_parser.rs (hunk 5 above)
Comment (AvaterClasher) [reply src/parser/exclusion_parser.rs:6]: Even if the user adds something weird I don't think that matters much as we are just matching the parsed result types against the excluded one's. So if they don't match they are naturally just ignored.

File: src/parser/exclusion_parser.rs (hunk 6 above)
Comment (AvaterClasher) [reply src/parser/exclusion_parser.rs:8]: Done in the previous commit

File: src/parser/params_as_comma_seperated.rs (hunk 3 above)
Comment (dteare) [reply src/parser/params_as_comma_seperated.rs:17]: Yes. I think it reads much better w/o them.

**Comment by AvaterClasher** [comment]
Hey @dteare,

I've addressed all of the Clippy warnings now. Regarding the error messages, I've included the filename along with the error. I plan to remove the `eprintln!` macro once I integrate the logging framework later on.

Could you please provide your suggestions for the README.md? Specifically, I'm thinking about the example section. Should I remove the upper part of it, or would you recommend it staying?

Thank you.

**Comment by dteare** [comment]
Some nice updates here. Thanks for continuing your work here @AvaterClasher. Closing in favour of PR #6, which includes all your commits with my edits applied on top.

**Pull request closed without merging**